R/pkgref.R

```diff
@@ -58,6 +58,11 @@
     grepl("/", pkg)
 }
 
+.is_bioc <- function(pkg,bioc_version){
```

Comment:
@chainsawriot This function requires the bioc version to check if the package is in a specific release. Couldnt find any other way to check this thus far, but this would require to also add bioc_version as a parameter to .normalize_pkg

Reply:
Suggestion for `.normalize_pkg()`: make bioc_version optional. If it is omitted, it is assumed to be a cran package. if present, check

```r
.normalize_pkg <- function(pkg,bioc_version) {
  if (pkg == "" || is.na(pkg)) {
    stop("Invalid `pkg`.", call. = FALSE)
  }
  if (isTRUE(.is_github(pkg))) {
    if (isTRUE(grepl("github\\.com", pkg))) {
      pkg <- .extract_github_handle(pkg)
    }
  }
  if (isTRUE(.is_pkgref(pkg))) {
    return(.clean_suffixes(pkg))
  }
  if (isTRUE(.is_github(pkg))) {
    return(paste0("github::", .clean_suffixes(pkg)))
  }
  if(missing(bioc_version)){
    return(paste0("cran::", .clean_suffixes(pkg)))
  } else{
    if(isTRUE(.is_bioc(pkg,bioc_version))){
      return(paste0("bioc::", .clean_suffixes(pkg)))
    } else{
      return(paste0("cran::", .clean_suffixes(pkg)))
    }
  }
}
```